### ISSUE TITLE
Some bugs [bsnes v108.1][Windows 10 1903]

### ISSUE THREAD

**Issue by Max833** [byuu/bsnes#31]
- When I want to change the hotkeys or input with double click, bsnes crashes: https://streamable.com/j61bd [**fixed**]

- there seems to be a big delay with the new list-style:
https://streamable.com/r1e41 

- "pause emulation when focus is lost" is now default and that's great, but when I open "System" "Settings" "Tools" or "Help", the framerate still drops a lot. It's probably a good idea to pause emulation there, too. :)

- games with enhancement chips still start when "Prefer HLE" is disabled

- filters don't work with normal PPU

Other than that, bsnes is working great. Thanks for this awesome emulator, byuu! :D

**Comment by byuu** [comment]
Hotkey mapping is a known issue, it's only been ported to GTK/Qt so far, it's a miracle it even compiles on Windows right now.

The listbox delay is indeed unfortunate. I'll try to improve it, but it may prove difficult.

Yeah I'll restore pause on modality for Windows. It's nice for consistency not to pause but in practice the performance is just dreadful.

I intended for it to fall back on HLE rather than not run. Prefer HLE means it'll disable LLE and always use HLE. The idea was for games to always run.

Filters indeed don't work with the pixel PPU, that PPU always outputs a pre-scaled image. Not much choice, but I can at least disable the menu when in pixel PPU mode I suppose.

**Comment by Max833** [comment]
Thank you for your quick reply, byuu!

> I intended for it to fall back on HLE rather than not run. Prefer HLE means it'll disable LLE and always use HLE. The idea was for games to always run.

I see, this makes sense. But can we at least have something like a message, so we know LLE is working?

![zhzh](https://user-images.githubusercontent.com/53529402/62229150-5f451880-b3bf-11e9-9299-0ffee526bca8.jpg)

**Comment by byuu** [comment]
I'll think about this, sure. I also want a way to indicate whether you're on the fast/accurate PPU/DSP, etc. I fixed the input crashing and allow all four input mappings on Windows now.

**Comment by jjfmz42015** [comment]
@byuu you are on fire with the updates today!!! I'm glad I have been lucky enough to get to experience your emulators. Thank you for what you do.

**Comment by jjfmz42015** [comment]
> 
> 
> I'll think about this, sure. I also want a way to indicate whether you're on the fast/accurate PPU/DSP, etc. I fixed the input crashing and allow all four input mappings on Windows now.


could always add a menu option called something along the lines of execution details that will open a window containing ppu mode, support files if loaded etc. or have details added to end of existing manifest

**Comment by Screwtapello** [comment]
This issue has a whole bunch of different issues in it, so it would be hard to keep track of when it's safe to be closed, so I've decided to split it up into individual issues. From the OP:

- "When I want to change the hotkeys..." is now fixed
- "there seems to be a big delay..." has been filed as #34 
- "when I open "System" "Settings" "Tools" or "Help"..." has been filed as #35 
- "games with enhancement chips still start..." has been rephrased based on byuu's comments, and filed as #36 
- "filters don't work with normal PPU" is working as designed, so I didn't file a new issue for it.

Thanks for reporting these issues @Max833, feel free to click the "Subscribe" button on all of these individual issues to keep up-to-date on what's happening with them.
